src/components/ui/search-expert.tsx

```diff
@@ -0,0 +1,106 @@
+"use client";
+import { usePathname, useRouter, useSearchParams } from "next/navigation";
+import type { Section } from "~/models/types";
+import { z } from "zod";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useForm } from "react-hook-form";
+import { Form, FormControl, FormField, FormItem} from "./form";
+import { Label } from "./label";
+import { Input } from "./input";
+import { Select, SelectContent, SelectItem, SelectTrigger } from "./select";
+import { SelectValue } from "@radix-ui/react-select";
+import { useEffect} from "react";
+
+
+const formSchema = z.object({
+  role: z.string(),
+  tech: z.string()
+})
+
+const ShowTechSearchWrapper = ({ roles } : { roles: Section[]}) => {
```

Comment:
Just a suggestion, but personally I like to have my component -and filename be the same. To still follow the naming conventions, which is kebab-case in this case, I would write the following as a file name:
show-tech-search-wrapper.tsx

I'll leave it up to you whether you want to change this or not.